ui/components/app/wallet-overview/coin-overview.tsx

```diff
@@ -142,25 +229,100 @@ export const CoinOverview = ({
-                  className={classnames(
-                    `${classPrefix}-overview__primary-balance`,
-                    {
-                      [`${classPrefix}-overview__cached-balance`]:
-                        balanceIsCached,
-                    },
-                  )}
-                  data-testid={`${classPrefix}-overview__primary-currency`}
-                  value={balanceToDisplay}
-                  type={PRIMARY}
-                  ethNumberOfDecimals={4}
-                  hideTitle
-                  shouldCheckShowNativeToken
-                  isAggregatedFiatOverviewBalance={
-                    !showNativeTokenAsMainBalance
-                  }
-                />
+                <>
+                  <Box onMouseEnter={handleMouseEnter} ref={setBoxRef}>
+                    <UserPreferencedCurrencyDisplay
+                      style={{ display: 'contents' }}
+                      className={classnames(
+                        `${classPrefix}-overview__primary-balance`,
+                        {
+                          [`${classPrefix}-overview__cached-balance`]:
+                            balanceIsCached,
+                        },
+                      )}
+                      data-testid={`${classPrefix}-overview__primary-currency`}
+                      value={balanceToDisplay}
+                      type={PRIMARY}
+                      ethNumberOfDecimals={4}
+                      hideTitle
+                      shouldCheckShowNativeToken
+                      isAggregatedFiatOverviewBalance={
+                        !showNativeTokenAsMainBalance && !isTestnet
+                      }
+                    />
+                  </Box>
+                  {shouldShowPopover &&
+                  !isTestnet &&
+                  !showNativeTokenAsMainBalance ? (
+                    <Popover
+                      referenceElement={referenceElement}
+                      isOpen={isOpen}
+                      position={PopoverPosition.BottomStart} // TODO check with design-team about this bottom start issue
+                      hasArrow
+                      flip
+                      backgroundColor={BackgroundColor.overlayAlternative} // TODO check with DS on this opacity issue
+                      className="balance-popover__container"
+                      padding={3}
+                      onClickOutside={handleClick}
+                      onPressEscKey={handleClick}
+                      preventOverflow
+                      style={{
+                        zIndex: 66,
+                      }}
+                    >
+                      <Box>
+                        <Box
+                          display={Display.Flex}
+                          justifyContent={JustifyContent.spaceBetween}
+                        >
+                          <Text
+                            variant={TextVariant.bodySmBold}
+                            color={TextColor.overlayInverse}
```

Comment:
Then we can remove these individual color delclarations
```suggestion
```

Reply:
fixed [05e181b](https://github.com/MetaMask/metamask-extension/pull/27161/commits/05e181ba23ac86dfe206593d64ae6bee843bf87d)